packages/components/data-table/src/cell.styles.js

```diff
@@ -104,7 +104,10 @@ const BaseCell = styled.td`
   position: relative;
   display: flex;
   background-color: ${vars.colorSurface};
-  border-bottom: 1px solid ${vars.colorNeutral90};
+  border-bottom: ${(props) =>
+    props.shouldRenderBottomBorder
```

Comment:
This is to make sure we don't have a double border on the bottom of the table (the one from the last row and the top one from the footer), replacing the negative 1px margin "trick".